client/libra-dev/src/account_resource.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use crate::{
+    data::{LibraAccountResource, LibraEventHandle, LibraStatus},
+    error::*,
+};
+use libra_types::{
+    account_config::AccountResource, account_state_blob::AccountStateBlob, event::EVENT_KEY_LENGTH,
+};
+use std::convert::TryFrom;
+use std::slice;
+
+pub fn libra_LibraAccountResource_from_safe(
+    blob: AccountStateBlob,
+) -> Result<LibraAccountResource, LibraStatus> {
+    clear_error();
+    match AccountResource::try_from(&blob) {
+        Ok(account_resource) => {
+            let mut authentication_key = [0u8; 32];
```

Comment:
authetication key is just a bytearray, but right now it's used as 32 bytes

Reply:
I assume there is some Rust side constant that defines the key sizes? I'd prefer us to use that so that if the key size changes, this code won't suddenly be incorrect.